Resources/ProxyProvider/ProxyProvider.py

```diff
@@ -0,0 +1,33 @@
+""" ProxyProvider base class for various proxy providers
+"""
```

Comment:
Maybe you add a few lines detailing what the actual proxy providers should implement?
IIUC the only method to implement is `getProxy`